src/classes/UTIL_Permissions_TEST.cls

```diff
@@ -47,6 +47,177 @@ public with sharing class UTIL_Permissions_TEST {
+            } catch (Exception ex) {
+                System.assert(ex instanceof UTIL_Permissions.InsufficientPermissionException,
+                    'Minimum access user should not have delete access to standard/custom object.');
+            }
+        }
+    }
+
+    @isTest
+    static void shouldReturnTrueForUserWithDeleteAccessToSObjectType() {
+        System.runAs(getSysAdmin()) {
+            UTIL_Permissions utilPermissions = UTIL_Permissions.getInstance();
+            System.assert(utilPermissions.canDelete(Contact.SObjectType),
+                'Privileged user should have delete access to standard object.');
+        }
+    }
+
+    @isTest
+    static void shouldReturnTrueForUserWithCreateAccessToFieldByDescribeFieldResult() {
+        System.runAs(getSysAdmin()) {
+            System.assert(UTIL_Permissions.canCreate(Contact.LastName.getDescribe()),
+                'Privileged user should have create access to standard field.');
+        }
+    }
+
+    @isTest
+    static void shouldThrowExceptionForUserWithoutCreateAccessToFieldByDescribeFieldResult() {
+        System.runAs(getReadOnlyUser()) {
+            try {
+                UTIL_Permissions.canCreate(Contact.LastName.getDescribe());
+            } catch (Exception ex) {
+                System.assert(ex instanceof UTIL_Permissions.InsufficientPermissionException,
+                    'Minimum access user should not have create access to standard field.');
+            }
+        }
+    }
+
+    @isTest
+    static void shouldReturnFalseForUserWithoutCreateAccessToFieldByString() {
+        System.runAs(getReadOnlyUser()) {
+            System.assert(!UTIL_Permissions.canCreate(standardObjectName, standardFieldName, false),
+                'Minimum access user should not have create access to standard field.');
+        }
+    }
+
+    @isTest
+    static void shouldReturnFalseForUserWithoutReadAccessToCustomFieldByString() {
+        System.runAs(getReadOnlyUser()) {
+            System.assert(!UTIL_Permissions.canRead(customObjectName, customFieldName, false),
+                'Minimum access user should not have read access to custom field.');
+        }
+    }
+
+    @isTest
+    static void shouldReturnFalseForUserWithoutUpdateAccessToFieldByString() {
+        System.runAs(getReadOnlyUser()) {
+            System.assert(!UTIL_Permissions.canUpdate(standardObjectName, standardFieldName, false),
+                'Minimum access user should not have update access to standard field.');
+        }
+    }
+
+    @isTest
+    static void shouldReturnFalseForUserWithoutDeleteAccessToCustomFieldByString() {
+        System.runAs(getReadOnlyUser()) {
+            System.assert(!UTIL_Permissions.canUpdate(customObjectName, customFieldName, false),
+                'Minimum access user should not have delete access to custom field.');
+        }
+    }
+
+    @isTest
+    static void shouldReturnTrueForUserWithoutCreateAccessAfterPermissionAssignment() {
```

Comment:
I feel like we're testing platform functionality with this one and can pull it. I checked and it doesn't seem to cover any lines that are not already covered by another test.

Reply:
Updated!